proxy/dropCookiesSelectively.js

```diff
@@ -0,0 +1,23 @@
+//@zaproxy-proxy
+
+globalCookies = ['CONSENT', 'GZ', 'SID']
+
+function proxyRequest(msg) {
+    globalCookies.forEach(function (cookieToDrop) {
+        cookies = msg.getRequestHeader().getHttpCookies() // This is a List<HttpCookie>
+        iterator = cookies.iterator()
+        while (iterator.hasNext()) {
+            cookie = iterator.next() // This is a HttpCookie
+            if (cookie.name == cookieToDrop) {
+                iterator.remove()
+                print('Stripped away: ' + cookie.name)
+            }
+        }
+        msg.getRequestHeader().setCookies(cookies)
```

Comment:
Per above comment, this could be called outside the function and only if there was any change (no need to re-set them if the same).